File: app/src/main/java/com/example/user/eventest/room/DateConverterDB.java
package com.example.user.eventest.room;

import android.annotation.SuppressLint;
import android.arch.persistence.room.TypeConverter;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by User on 24.01.2018.
 */

public class DateConverterDB {
    private static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss";
    @SuppressLint("SimpleDateFormat")
    private DateFormat dateFormat = new SimpleDateFormat(DATE_FORMAT);

    @TypeConverter
    public Date dateFromString(String value) {
        try {
            return value == null ? null : dateFormat.parse(value);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return null;
    }

    @TypeConverter
    public String stringFromDate(Date date) {

        if (date == null) {
            return null;
        } else {
            return dateFormat.format(date);
        }
    }
}

File: app/src/main/java/com/example/user/eventest/room/DateConverterDB.kt
package com.example.user.eventest.room

import android.annotation.SuppressLint
import android.arch.persistence.room.TypeConverter
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by DR
 * on 24.01.2018.
 */
const val DATE_FORMAT = "yyyy-MM-dd HH:mm:ss"

class DateConverterDB {
    @SuppressLint("SimpleDateFormat")
    private val dateFormat = SimpleDateFormat(DATE_FORMAT)

    @TypeConverter
    fun dateFromString(value: String): Date = dateFormat.parse(value)

    @TypeConverter
    fun stringFromDate(date: Date): String = dateFormat.format(date)
}
